RoundCheckbox.js

```diff
@@ -27,19 +32,90 @@ export default class RoundCheckbox extends React.PureComponent {
     onValueChange: () => {},
   };
 
+  state = {
+    scaleAndOpacityOfCheckbox: null,
+  };
+
+  componentDidMount() {
+    this.setState({
+      scaleAndOpacityOfCheckbox: new Animated.Value(this.props.checked ? 1 : 0),
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.checked !== prevProps.checked) {
+      console.log(this.state.scaleAndOpacityOfCheckbox);
+      if (this.props.checked) {
+        Animated.timing(this.state.scaleAndOpacityOfCheckbox, {
+          toValue: 1,
+          duration: 300,
+          useNativeDriver: false,
+        }).start();
+      } else {
+        Animated.timing(this.state.scaleAndOpacityOfCheckbox, {
+          toValue: 0,
+          duration: 300,
+          useNativeDriver: false,
+        }).start();
+      }
+    }
+  }
+
   render() {
     const iconSize = parseInt(this.props.size * 1.3);
+
+    const bothStyles = {
+      width: this.props.size,
+      height: this.props.size,
+
+      borderRadius: this.props.size / 2,
+    };
+
+    const checkedStyles = {
+      backgroundColor: this.props.backgroundColor,
+      borderColor: this.props.backgroundColor,
+      opacity: this.state.scaleAndOpacityOfCheckbox,
+      scaleX: this.state.scaleAndOpacityOfCheckbox,
+      scaleY: this.state.scaleAndOpacityOfCheckbox,
+      position: 'absolute',
+      top: 0,
+      left: 0,
+    };
+
+    const uncheckedStyles = {
+      backgroundColor: 'transparent',
+      borderColor: this.props.borderColor,
+    };
+
     return (
-      <TouchableWithoutFeedback hitSlop={hitSlop} onPress={this._onPress}>
-        <View
-          shouldRasterizeIOS={true}
-          style={[this.getIconWrapperStyle(), styles.commonWrapperStyles, this.props.style]}
-        >
-          <Icon
-            name={this.props.icon}
-            color={this.props.checked ? this.props.iconColor : 'transparent'}
-            style={{ height: iconSize, fontSize: iconSize, backgroundColor: 'transparent' }}
-          />
+      <TouchableWithoutFeedback hitSlop={hitSlop} onPress={this._onPress} style={this.props.style}>
+        <View style={[styles.parentWrapper]}>
+          <View
+            shouldRasterizeIOS={true}
+            style={[uncheckedStyles, bothStyles, styles.commonWrapperStyles]}>
+            <Icon
```

Comment:
why do we need to render this icon? wouldn't rendering one icon be enough?

Reply:
We need one icon (the unchecked one) to stay on screens at all times for the animation to look proper. The other icon simply renders on top of it, no problems.

Reply:
> We need one icon (the unchecked one) to stay on screens at all times for the animation to look proper.

sorry can you expand on this? What exactly doesn't look right? Thanks